src/Symfony/Component/Yaml/Tests/DumperTest.php

```diff
@@ -583,27 +587,34 @@ public function testDumpMultiLineStringAsScalarBlockWhenFirstLineHasLeadingSpace
 
         $expected = "data:\n    multi_line: |4-\n            the first line has leading spaces\n        The second line does not.";
 
-        $this->assertSame($expected, $this->dumper->dump($data, 2, 0, Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK));
+        $yml = $this->dumper->dump($data, 2, 0, Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK);
+        $this->assertSame($expected, $yml);
+        $this->assertSame($data, $this->parser->parse($yml));
     }
 
     public function testCarriageReturnFollowedByNewlineIsMaintainedWhenDumpingAsMultiLineLiteralBlock()
     {
-        $this->assertSame("- \"a\\r\\nb\\nc\"\n", $this->dumper->dump(["a\r\nb\nc"], 2, 0, Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK));
+        $data = ["a\r\nb\nc"];
+        $expected = "- \"a\\r\\nb\\nc\"\n";
+        $this->assertSame($expected, $this->dumper->dump($data, 2, 0, Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK));
+        $this->assertSame($data, $this->parser->parse($expected));
     }
 
     public function testCarriageReturnNotFollowedByNewlineIsPreservedWhenDumpingAsMultiLineLiteralBlock()
     {
+        $data = [
+            'parent' => [
+                'foo' => "bar\n\rbaz: qux",
+            ],
+        ];
         $expected = <<<'YAML'
 parent:
     foo: "bar\n\rbaz: qux"
 
 YAML;
-
-        $this->assertSame($expected, $this->dumper->dump([
-            'parent' => [
-                'foo' => "bar\n\rbaz: qux",
-            ],
-        ], 4, 0, Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK));
+        $yaml = $this->dumper->dump($data, 4, 0, Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK);
+        $this->assertSame($expected, $yaml);
+        $this->assertSame($data, Yaml::parse($yaml));
```

Comment:
why not reusing `$this->parser` in this test compared to the other ones ?

Reply:
Copied from elsewhere. It was already inconsistent to begin with.
I am now fixing it everywhere in the class.